lms/djangoapps/learner_dashboard/tests/test_views.py

```diff
@@ -0,0 +1,105 @@
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        modulestore_course = ModuleStoreCourseFactory()
+        course_run = CourseRunFactory(key=str(modulestore_course.id))
+        course = CourseFactory(course_runs=[course_run])
+        cls.program = ProgramFactory(uuid=cls.program_uuid, courses=[course])
+
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        self.client.login(username=self.user.username, password=self.password)
+        self.set_program_in_catalog_cache(self.program_uuid, self.program)
+        ProgramEnrollmentFactory.create(
+            user=self.user,
+            program_uuid=self.program_uuid,
+            external_user_key='0001',
+        )
+
+    @override_waffle_flag(ENABLE_PROGRAM_DISCUSSIONS, active=True)
+    def test_program_discussion_not_configured(self):
+        """
+        Verify API returns proper response in case ProgramDiscussions is not Configured.
+        """
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, {'iframe': None, 'enabled': True})
+
+    @override_waffle_flag(ENABLE_PROGRAM_DISCUSSIONS, active=True)
+    def test_if_user_is_not_authenticated(self):
+        """
+        Verify that 401 is returned if user is not authenticated.
+        """
+        self.client.logout()
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 401)
+
+    @override_waffle_flag(ENABLE_PROGRAM_DISCUSSIONS, active=True)
+    def test_api_returns_discussions_iframe(self):
+        """
+        Test if API returns iframe in case ProgramDiscussionsConfiguration model contains proper data
+        """
+        discussion_config = ProgramDiscussionsConfiguration.objects.create(
+            program_uuid=self.program_uuid,
+            enabled=True,
+            provider_type="lti_provider",
+        )
+        discussion_config.lti_configuration = LtiConfiguration.objects.create(
+            config_store=LtiConfiguration.CONFIG_ON_DB,
+            lti_1p1_launch_url='http://test.url',
+            lti_1p1_client_key='test_client_key',
+            lti_1p1_client_secret='test_client_secret',
+        )
+        discussion_config.save()
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertIsInstance(response.data['iframe'], Markup)
+        self.assertIn('iframe', str(response.data['iframe']),)
+
+    @override_waffle_flag(ENABLE_PROGRAM_DISCUSSIONS, active=True)
+    def test_program_does_not_exist(self):
+        """
+        Test if API returns 404 in case program does not exist
+        """
+        response = self.client.get(reverse('program_discussion', kwargs={'program_uuid': str(uuid4())}))
+        self.assertEqual(response.status_code, 404)
+
+    @override_waffle_flag(ENABLE_PROGRAM_DISCUSSIONS, active=True)
```

Comment:
you should override `@override_waffle_flag(ENABLE_PROGRAM_DISCUSSIONS, active=True)` on the class level because most of the tests depend on this value. 
this will refactor the duplicate code.